### ISSUE TITLE
Disable Elasticsearch health check in development

### ISSUE THREAD

**Issue by mraible** [jhipster/generator-jhipster#9407]
##### **Overview of the feature request**

When using the `dev` profile, the Elastisearch health check should be disabled. Otherwise, developers will see the following stack trace and pester us incessantly with bug reports. Proof: I've already done it on the mailing list!

##### **Motivation for or Use Case**

It's annoying and makes you think something is wrong. 

```
2019-03-08 20:50:14.555  WARN 12188 --- [nfoReplicator-0] s.b.a.e.ElasticsearchRestHealthIndicator : Elasticsearch health check failed

java.net.ConnectException: Connection refused
        at org.elasticsearch.client.RestClient$SyncResponseListener.get(RestClient.java:943)
        at org.elasticsearch.client.RestClient.performRequest(RestClient.java:227)
        at org.springframework.boot.actuate.elasticsearch.ElasticsearchRestHealthIndicator.doHealthCheck(ElasticsearchRestHealthIndicator.java:61)
        at org.springframework.boot.actuate.health.AbstractHealthIndicator.health(AbstractHealthIndicator.java:84)
        at org.springframework.boot.actuate.health.CompositeHealthIndicator.health(CompositeHealthIndicator.java:98)
        at org.springframework.cloud.netflix.eureka.EurekaHealthCheckHandler.getHealthStatus(EurekaHealthCheckHandler.java:103)
        at org.springframework.cloud.netflix.eureka.EurekaHealthCheckHandler.getStatus(EurekaHealthCheckHandler.java:99)
        at com.netflix.discovery.DiscoveryClient.refreshInstanceInfo(DiscoveryClient.java:1388)
        at com.netflix.discovery.InstanceInfoReplicator.run(InstanceInfoReplicator.java:117)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
Caused by: java.net.ConnectException: Connection refused
```

**Comment by pascalgrimaud** [comment]
I got the same problem when reviewing the Kubernetes PR.

It should be resolved now.
See my https://github.com/jhipster/generator-jhipster/pull/9365#issuecomment-479939915 and my commit https://github.com/jhipster/generator-jhipster/pull/9365/commits/49f9cb39792a902c40250e2cfb41cb7f789ac8b2

I couldn't reproduce any more. Don't hesitate to reopen if you encounter this issue again, Matt.

**Comment by alexsintoni** [comment]
I guys,
I got the same error with the last JHipster version v6.0.0.
I would like to re-open this ticket but I don't know how.

Anyway, following my configuration:
```
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "com.company",
      "nativeLanguage": "en"
    },
    "jhipsterVersion": "6.0.0",
    "applicationType": "uaa",
    "baseName": "UserAuthorityAgent",
    "packageName": "com.kekse.mystaff",
    "packageFolder": "com/kekse/mystaff",
    "serverPort": "9999",
    "authenticationType": "uaa",
    "cacheProvider": "hazelcast",
    "enableHibernateCache": true,
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "mysql",
    "prodDatabaseType": "mysql",
    "searchEngine": "elasticsearch",
    "messageBroker": false,
    "serviceDiscoveryType": "eureka",
    "buildTool": "maven",
    "enableSwaggerCodegen": false,
    "jwtSecretKey": "",
    "testFrameworks": [
      "gatling"
    ],
    "jhiPrefix": "jhi",
    "entitySuffix": "",
    "dtoSuffix": "DTO",
    "otherModules": [],
    "enableTranslation": true,
    "clientPackageManager": "npm",
    "nativeLanguage": "en",
    "languages": [
      "en",
      "al",
      "ar-ly",
      "hy",
      "by",
      "bn",
      "ca",
      "zh-cn",
      "zh-tw",
      "cs",
      "da",
      "nl",
      "et",
      "fa",
      "fr",
      "gl",
      "de",
      "el",
      "hi",
      "hu",
      "id",
      "it",
      "ja",
      "ko",
      "mr",
      "my",
      "pl",
      "pt-br",
      "pt-pt",
      "ro",
      "ru",
      "sk",
      "sr",
      "es",
      "sv",
      "tr",
      "ta",
      "te",
      "th",
      "ua",
      "uz-cyr",
      "uz-lat",
      "vi"
    ],
    "skipClient": true
  }
}
```

I have the same error in all: UAA, Application and Gateway.
All v6.0.0

**Comment by pascalgrimaud** [comment]
@alexsintonin : did you play with config ?
See https://github.com/jhipster/generator-jhipster/commit/49f9cb39792a902c40250e2cfb41cb7f789ac8b2

Can you try:

```yaml
spring:
    elasticsearch:
        rest:
            uris: http://localhost:9200
```

**Comment by alexsintoni** [comment]
Hi @pascalgrimaud ,

I added that configuration to `application-dev.yml` as you can see below, but still same error:
```
# ===================================================================
# Spring Boot configuration for the "dev" profile.
#
# This configuration overrides the application.yml file.
#
# More information on profiles: https://www.jhipster.tech/profiles/
# More information on configuration properties: https://www.jhipster.tech/common-application-properties/
# ===================================================================

# ===================================================================
# Standard Spring Boot properties.
# Full reference is available at:
# http://docs.spring.io/spring-boot/docs/current/reference/html/common-application-properties.html
# ===================================================================

logging:
  level:
    ROOT: DEBUG
    io.github.jhipster: DEBUG
    com.kekse.mystaff: DEBUG

eureka:
  instance:
    prefer-ip-address: true
  client:
    service-url:
      defaultZone: http://admin:${jhipster.registry.password}@${DEV_REGISTRY_ADDRESS}:${DEV_REGISTRY_PORT}/eureka/

spring:
  elasticsearch:
    rest:
      uris: http://localhost:9200
  profiles:
    active: dev
    include:
      - swagger
      # Uncomment to activate TLS for the dev profile
      #- tls
  devtools:
    restart:
      enabled: true
    livereload:
      enabled: false # we use Webpack dev server + BrowserSync for livereload
  jackson:
    serialization:
      indent-output: true
  datasource:
    type: com.zaxxer.hikari.HikariDataSource
    url: jdbc:mysql://${DEV_DB_ADDRESS}:${DEV_DB_PORT}/UserAuthorityAgent600?createDatabaseIfNotExist=true&useUnicode=true&characterEncoding=utf8&useSSL=false&useLegacyDatetimeCode=false&serverTimezone=UTC
    username: ${DEV_DB_USERNAME}
    password: ${DEV_DB_PASSWORD}
    hikari:
      poolName: Hikari
      auto-commit: false
      data-source-properties:
        cachePrepStmts: true
        prepStmtCacheSize: 250
        prepStmtCacheSqlLimit: 2048
        useServerPrepStmts: true
  jpa:
    database-platform: org.hibernate.dialect.MySQL5InnoDBDialect
    database: MYSQL
    show-sql: true
    properties:
      hibernate.id.new_generator_mappings: true
      hibernate.connection.provider_disables_autocommit: true
      hibernate.cache.use_second_level_cache: true
      hibernate.cache.use_query_cache: false
      hibernate.generate_statistics: false
      hibernate.cache.region.factory_class: com.hazelcast.hibernate.HazelcastCacheRegionFactory
      hibernate.cache.hazelcast.instance_name: UserAuthorityAgent
      hibernate.cache.use_minimal_puts: true
      hibernate.cache.hazelcast.use_lite_member: true
  data:
    elasticsearch:
      properties:
        path:
          home: target/elasticsearch
  liquibase:
    # Remove 'faker' if you do not want the sample data to be loaded automatically
    contexts: dev, faker
  mail:
    host: ${DEV_MAIL_HOST}
    port: ${DEV_MAIL_PORT}
    username: ${DEV_MAIL_USERNAME}
    password: ${DEV_MAIL_PASSWORD}
    protocol: ${DEV_MAIL_PROTOCOL}
    tls: true
    properties.mail.smtp:
      auth: true
      starttls.enable: true
      ssl.trust: ${DEV_MAIL_SSL_TRUST}
  messages:
    cache-duration: PT1S # 1 second, see the ISO 8601 standard
  thymeleaf:
    cache: false
  sleuth:
    sampler:
      probability: 1 # report 100% of traces
  zipkin: # Use the "zipkin" Maven profile to have the Spring Cloud Zipkin dependencies
    base-url: http://localhost:9411
    enabled: false
    locator:
      discovery:
        enabled: true

server:
  port: 9999

# ===================================================================
# JHipster specific properties
#
# Full reference is available at: https://www.jhipster.tech/common-application-properties/
# ===================================================================

jhipster:
  cache: # Cache configuration
    hazelcast: # Hazelcast distributed cache
      time-to-live-seconds: 3600
      backup-count: 1
      management-center: # Full reference is available at: http://docs.hazelcast.org/docs/management-center/3.9/manual/html/Deploying_and_Starting.html
        enabled: false
        update-interval: 3
        url: http://localhost:8180/mancenter
  # CORS is only enabled by default with the "dev" profile, so BrowserSync can access the API
  cors:
    allowed-origins: '*'
    allowed-methods: '*'
    allowed-headers: '*'
    exposed-headers: 'Authorization,Link,X-Total-Count'
    allow-credentials: true
    max-age: 1800
  security:
    client-authorization:
      client-id: internal
      client-secret: internal
  mail: # specific JHipster mail property, for standard properties see MailProperties
    from: UserAuthorityAgent@localhost
    base-url: http://127.0.0.1:8080
  metrics:
    logs: # Reports metrics in the logs
      enabled: false
      report-frequency: 60 # in seconds
  logging:
    use-json-format: false # By default, logs are not in Json format
    logstash: # Forward logs to logstash over a socket, used by LoggingConfiguration
      enabled: false
      host: localhost
      port: 5000
      queue-size: 512
uaa:
  key-store:
    name: config/tls/keystore.p12
    password: password
    alias: selfsigned
  web-client-configuration:
    # Access Token is valid for 5 mins
    access-token-validity-in-seconds: 300
    # Refresh Token is valid for 7 days
    refresh-token-validity-in-seconds-for-remember-me: 604800
    client-id: web_app
    secret: changeit
# ===================================================================
# Application specific properties
# Add your own application properties here, see the ApplicationProperties class
# to have type-safe configuration, like in the JHipsterProperties above
#
# More documentation is available at:
# https://www.jhipster.tech/common-application-properties/
# ===================================================================

# application:

```

Elasticsearch health check failed:
```
2019-05-08 18:08:46.481  WARN 6724 --- [nfoReplicator-0] s.b.a.e.ElasticsearchRestHealthIndicator : Elasticsearch health check failed

java.net.ConnectException: Connection refused
	at org.elasticsearch.client.RestClient$SyncResponseListener.get(RestClient.java:943)
	at org.elasticsearch.client.RestClient.performRequest(RestClient.java:227)
	at org.springframework.boot.actuate.elasticsearch.ElasticsearchRestHealthIndicator.doHealthCheck(ElasticsearchRestHealthIndicator.java:61)
	at org.springframework.boot.actuate.health.AbstractHealthIndicator.health(AbstractHealthIndicator.java:84)
	at org.springframework.boot.actuate.health.CompositeHealthIndicator.health(CompositeHealthIndicator.java:98)
	at org.springframework.cloud.netflix.eureka.EurekaHealthCheckHandler.getHealthStatus(EurekaHealthCheckHandler.java:110)
	at org.springframework.cloud.netflix.eureka.EurekaHealthCheckHandler.getStatus(EurekaHealthCheckHandler.java:106)
	at com.netflix.discovery.DiscoveryClient.refreshInstanceInfo(DiscoveryClient.java:1388)
	at com.netflix.discovery.InstanceInfoReplicator.run(InstanceInfoReplicator.java:117)
	at com.netflix.discovery.InstanceInfoReplicator$1.run(InstanceInfoReplicator.java:101)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.net.ConnectException: Connection refused
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:717)
	at org.apache.http.impl.nio.reactor.DefaultConnectingIOReactor.processEvent(DefaultConnectingIOReactor.java:171)
	at org.apache.http.impl.nio.reactor.DefaultConnectingIOReactor.processEvents(DefaultConnectingIOReactor.java:145)
	at org.apache.http.impl.nio.reactor.AbstractMultiworkerIOReactor.execute(AbstractMultiworkerIOReactor.java:351)
	at org.apache.http.impl.nio.conn.PoolingNHttpClientConnectionManager.execute(PoolingNHttpClientConnectionManager.java:221)
	at org.apache.http.impl.nio.client.CloseableHttpAsyncClientBase$1.run(CloseableHttpAsyncClientBase.java:64)
	... 1 common frames omitted
```

**Comment by pascalgrimaud** [comment]
@alexsintoni : just tried your config and I don't have this stacktrace about health check. I generated your UAA application, then:
- start the registry with docker-compose
- start MySQL with docker-compose
- start Elasticsearch with docker-compose
- start your UAA app, using `./mvnw`

Did you start your local Elasticsearch ? Then be sure it is available at `http://localhost:9200`. For me it's the case, as I am with Ubuntu

See my screenshot with `docker ps -a` :

![Capture d’écran de 2019-05-08 18-25-47](https://user-images.githubusercontent.com/9156882/57391580-19f9bc00-71bf-11e9-82a9-3526680cb16a.png)

**Comment by mraible** [comment]
@pascalgrimaud When I originally opened this issue, it was because it happened when Elasticsearch wasn't running with Docker Compose. When you're running the `dev` profile, Elasticsearch runs in embedded mode and it uses the local file system instead of a URL.

**Comment by pascalgrimaud** [comment]
@mraible : oh I see. Not sure if there is an embedded Elasticsearch or not. We need to check this

**Comment by mraible** [comment]
There is, using [Spring Data Jest](https://github.com/VanRoy/spring-data-jest). I [coded it](https://github.com/jhipster/generator-jhipster/pull/8122) myself. The reason I opened this ticket initially is because there's still a health check and it should be disabled somehow when running in `dev` mode.

**Comment by pascalgrimaud** [comment]
Ok, I understand better the problem.
When working on Kubernetes, I got the same stacktrace with healthcheck in the logs, I thought it came from this property.

For disable the health check for Elasticsearch in dev, we can use, in `application-dev.yml` :
```
management.health.elasticsearch.enabled: false
```

Can you tell me if this works for you @alexsintoni ?
I'll probably PR this change

**Comment by alexsintoni** [comment]
Hi @pascalgrimaud ,

really thanks for your help.
I was not aware that from JHipster v6.0.0 I need to start Elasticsearch with docker-compose. All previous versions I could always start all the micro-services with `./mvnw`.

What command should I use to start Elasticsearch?
I also guess that I have to start Elasticsearch only once and it will be used by UAA, My Micro-Service Application and the Gateway. 
So I need to add the following config to all of them:
```
spring:
  elasticsearch:
    rest:
      uris: http://localhost:9200
```

Is it correct?

Thanks in advance for your help :)

**Comment by pascalgrimaud** [comment]
No, i was wrong !! You don't need to start elasticsearch in dev
There is an embedded Elasticsearch. 

The fix should be this one: 
https://github.com/jhipster/generator-jhipster/pull/9715
